File: viperaveil-main/Discord-ELO/viperaelov2/snekwarselo/snekwarselo/Commands/MatchCheck.cs
using DSharpPlus;
using DSharpPlus.CommandsNext;
using DSharpPlus.Entities;
using DSharpPlus.SlashCommands;
using Newtonsoft.Json;
using snekwarselo.Dtos;
using snekwarselo.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace snekwarselo.Commands
{
    public static class MatchCheck
    {
        public static string Uri { get; set; }

        private static async Task<bool> CheckHasMatch()
        {
            string hasMatchUrl = $"{Uri}/has-match";
            using (HttpClient httpClient = new HttpClient())
            {
                HttpResponseMessage response = await httpClient.GetAsync(hasMatchUrl);
                if (response.IsSuccessStatusCode)
                {
                    string jsonResponse = await response.Content.ReadAsStringAsync();
                    return JsonConvert.DeserializeObject<bool>(jsonResponse);
                }
            }
            return false;
        }

        private static async Task<string> GetMatch()
        {
            string getMatchUrl = $"{Uri}/get-match";
            using (HttpClient httpClient = new HttpClient())
            {
                HttpResponseMessage response = await httpClient.GetAsync(getMatchUrl);
                if (response.IsSuccessStatusCode)
                {
                    return await response.Content.ReadAsStringAsync();
                }
            }
            return null;
        }

        private static MatchDto DeserializeMatch(string matchJson)
        {
            // Add your custom deserialization logic here
            return JsonConvert.DeserializeObject<MatchDto>(matchJson);
        }

        private static async void ProcessMatch(MatchDto matchData, DiscordClient sender)
        {
            // Add your custom processing logic here
            // Match Announcement channel
            var channel = await sender.GetChannelAsync(1109347736135933952);
            List<string> team1 = new List<string>();
            List<string> team2 = new List<string>();
            foreach(PlayerDto player in matchData.Team1)
            {
                team1.Add(player.Name);
            }
            foreach (PlayerDto player in matchData.Team2)
            {
                team2.Add(player.Name);
            }
            var redTeam = string.Join(",", team1);
            var blueTeam = string.Join(",", team2);

            DiscordEmbedBuilder embed = new DiscordEmbedBuilder()
            .WithColor(DiscordColor.Green)
            .WithTitle("Snek Wars Match")
            .WithDescription($"New Match has been found for game mode")
            .AddField("Red Team", redTeam)
            .AddField("Blue Team", blueTeam);

            DiscordMessageBuilder messageBuilder = new DiscordMessageBuilder().AddEmbed(embed)
                .AddComponents();

            await channel.SendMessageAsync(messageBuilder);
            Console.WriteLine("Match found!");
            Console.WriteLine(matchData);
            // ...
        }

        public static async Task MatchCheckLoop(DiscordClient sender, DSharpPlus.EventArgs.ReadyEventArgs e)
        {
            while (true)
            {
                bool hasMatch = await CheckHasMatch();
                if (hasMatch)
                {
                    string match = await GetMatch();
                    if (!string.IsNullOrEmpty(match))
                    {
                        MatchDto matchData = DeserializeMatch(match);
                        ProcessMatch(matchData,sender);
                    }
                }
                Thread.Sleep(5000);
            }
        }
    }
}

File: viperaveil-main/Discord-ELO/viperaelo/src/commands/matchcheck.py
import discord
import random
from discord import Embed
from discord.ext import commands
import time
import json
import requests
from src.utils.constants import getBaseUrlEndpoint

# Class watches the match cache 
class MatchCheckCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot
        self.api_base_url = f"{getBaseUrlEndpoint()}/api/matchmaking"  # Replace with the actual base URL of your API

    @commands.Cog.listener()
    async def on_ready(self):
        print("Bot is ready.")
        self.bot.loop.create_task(self.match_check_loop())

    async def check_has_match(self):
        has_match_url = f"{self.api_base_url}/has-match"
        response = requests.get(has_match_url)
        if response.status_code == 200:
            return response.json()
        else:
            return None

    async def get_match(self):
        get_match_url = f"{self.api_base_url}/get-match"
        response = requests.get(get_match_url)
        if response.status_code == 200:
            return response.json()
        else:
            return None

    def deserialize_match(self, match_json):
        # Add your custom deserialization logic here
        match_data = json.loads(match_json)
        return match_data

    def process_match(self, match_data):
        # Add your custom processing logic here
        print("Match found!")
        print(match_data)
        # ...

    async def match_check_loop(self):
        while True:
            has_match = await self.check_has_match()
            if has_match:
                match = await self.get_match()
                if match:
                    match_data = self.deserialize_match(match)
                    self.process_match(match_data)
            time.sleep(5)
